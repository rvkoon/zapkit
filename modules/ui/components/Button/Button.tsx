import { MouseEvent, ReactNode } from "react";
import cx from "classnames";

type ButtonTypes = "button" | "submit";
export interface IButtonProps {
  content: string | ReactNode;
  onClick?: () => void;
  className?: string;
  type?: ButtonTypes;
  disabled?: boolean;
  testId?: string;
  id?: string;
}

export function Button({
  content,
  onClick,
  className,
  type = "button",
  disabled,
  testId,
  id,
}: IButtonProps): JSX.Element {
  function handleClick(event: MouseEvent<HTMLButtonElement>) {
    if (onClick) {
      event.preventDefault();
      onClick();
    }
  }

  return (
    <button
      id={id}
      type={type}
      className={cx(
        "min-h-4 py-1 px-2 bg-primary-blue hover:bg-primary-blue-light text-white font-bold text-[14px] rounded-base",
        disabled && "pointer-none opacity-30",
        className
      )}
      onClick={handleClick}
      disabled={disabled}
      data-testid={testId}
    >
      {content}
    </button>
  );
}
