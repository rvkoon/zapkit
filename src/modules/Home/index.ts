export * from './components';
export * from './helpers';
export * from './hooks';
export * from './services';
export * from './types';
export * from './constants';