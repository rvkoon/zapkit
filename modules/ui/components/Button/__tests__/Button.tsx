import { Button, IButtonProps } from "../Button";
import { RenderResult } from "@testing-library/react";
import { render } from "modules/tests";

describe("Button component", () => {
  function renderComponent(props: IButtonProps): RenderResult {
    return render(<Button {...props} />);
  }

  it("should render correctly", () => {
    const { container } = renderComponent({ content: "Click me" });
    expect(container).toMatchSnapshot();
  });
});
