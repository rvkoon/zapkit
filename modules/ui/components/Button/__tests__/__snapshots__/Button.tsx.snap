// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly 1`] = `
<div>
  <button
    class="min-h-4 py-1 px-2 bg-primary-blue hover:bg-primary-blue-light text-white font-bold text-[14px] rounded-base"
    type="button"
  >
    Click me
  </button>
</div>
`;
